pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/PartitionSegmentPruner.java

```diff
@@ -79,18 +90,30 @@ public boolean pruneSegment(FilterQueryTree filterQueryTree, Map<String, ColumnM
 
     // Leaf node
     String column = filterQueryTree.getColumn();
-    ColumnMetadata metadata = columnMetadataMap.get(column);
-    if (metadata == null) {
-      return false;
+    List<IntRange> partitionRanges;
```

Comment:
Refactor this out so that the high level recursive function is concise and clear to read.